File: content/api-documentation/how-to/orders/examples/streaming.js
const Alpaca = require('@alpacahq/alpaca-trade-api')
const alpaca = new Alpaca()

// Prepare the websocket connection and subscribe to trade_updates.
alpaca.websocket.onConnect(function () {
    console.log("Connected")
    client.subscribe(['trade_updates'])
    setTimeout(() => {
        client.disconnect()
    }, 30 * 1000)
})
alpaca.websocket.onDisconnect(() => {
    console.log("Disconnected")
})
alpaca.websocket.onStateChange(newState => {
    console.log(`State changed to ${newState}`)
})

// Handle updates on an order you've given a Client Order ID.
const client_order_id = 'my_client_order_id'
alpaca.websocket.onOrderUpdate(orderData => {
    const event = orderData['event']
    if (orderData['order']['client_order_id'] == client_order_id) {
        console.log(`Update for ${client_order_id}. Event: ${event}.`)
    }
})

// Start listening for updates.
alpaca.websocket.connect()

File: content/api-documentation/how-to/orders/examples/streaming.go
package main

import (
	"fmt"

	"github.com/alpacahq/alpaca-trade-api-go/alpaca"
    "github.com/alpacahq/alpaca-trade-api-go/stream"
)

func init() {
	alpaca.SetBaseUrl("https://paper-api.alpaca.markets")
}

func main() {
	if err := stream.Register(alpaca.TradeUpdates, tradeHandler); err != nil {
        panic(err)
    }
}

func tradeHandler(msg interface{}) {
	tradeupdate := msg.(alpaca.TradeUpdate)
	if tradeupdate.Order.ClientOrderID == "my_client_order_id" {
		fmt.Printf("Update for {}. Event: {}.\n", tradeupdate.Order.ClientOrderID, tradeupdate.Event)
	}
}